csharp/src/Ice/InputStream.cs

```diff
@@ -201,6 +191,45 @@ public void EndEncapsulation()
             ResetEncapsulation();
         }
 
+        /// <summary>Restarts the most recently started encapsulation.</summary>
+        public void RestartEncapsulation()
+        {
+            if (_mainEncapsBackup == null)
+            {
+                throw new EncapsulationException();
+            }
+
+            if (_mainEncaps != null || _endpointEncaps != null)
+            {
+                ResetEncapsulation();
+            }
+
+            // Restore backup:
+            _mainEncaps = _mainEncapsBackup.Value.Encaps;
+            Pos = _mainEncapsBackup.Value.Pos;
+            Encoding = _mainEncapsBackup.Value.Encoding;
+            _minTotalSeqSize = _mainEncapsBackup.Value.MinTotalSeqSize;
+            _limit = Pos + _mainEncaps.Value.Size - 6;
+        }
+
+        /// <summary>Verifies if this InputStream can read data encoded using its current encoding.
+        /// Throws Ice.UnsupportedEncodingException if it cannot.</summary>
+        public void CheckIsReadable()
```

Comment:
It could be named CheckSupportedEncoding :)

Reply:
I find Readable clearer than Supported here. This method tells you whether or the not the InputStream (that typically holds a request frame or a response frame) can decode data encoded in its current encoding.  Even if the InputStream can't decode this encoding, it's still "supported" in the sense that Ice can forward a payload with this encoding ... and long as there is no need to read it.